src/envoy/http/authn/authn_utils.cc

```diff
@@ -36,32 +36,41 @@ bool JsonIterCallback::JsonIteratorCallback(const std::string& key,
   ENVOY_LOG(debug, "{}: key is {}", __FUNCTION__, key);
   ::google::protobuf::Map< ::std::string, ::std::string>* claims =
       payload->mutable_claims();
+
+  // In current implementation, only string objects are extracted into
+  // claims. If call obj.asJsonString(), will get "panic: not reached" from
+  // json_loader.cc.
   try {
```

Comment:
Keep try-catch block short, and avoid nested. Simplify this code to smt like:

try {
  std::string obj_str = obj.asString();
  (*claims)[key] = obj_str;
} catch (Json::Exception&) {
  // Ignore non-string claims.
}

if (key == "aud") {
  if (claims.find(key) != claims.end()) {
    payload->add_audiences(claims[key]);
  } else {
    try {
       aud_vector = obj.getStringArray(key, true);
       ....
    } catch (Json::Exception& e) {
      ENVOY_LOG(error, "aud field type is not string nor string array: {}", e.what());
    }
  }

Reply:
Done.